src/dialects/db2/data-types.ts

```diff
@@ -0,0 +1,380 @@
+    || dataType._unsigned
+    || dataType._zerofill
+  ) {
+    warn(
+      `Db2 does not support '${dataType.key}' with options. Plain '${dataType.key}' will be used instead.`,
+    );
+    dataType._length = undefined;
+    dataType.options.length = undefined;
+    dataType._unsigned = undefined;
+    dataType._zerofill = undefined;
+  }
+}
+
+/**
+ * types: [hex, ...]
+ *
+ * @see Data types and table columns: https://www.ibm.com/support/knowledgecenter/en/SSEPGG_11.1.0/com.ibm.db2.luw.admin.dbobj.doc/doc/c0055357.html
+ */
+
+BaseTypes.DATE[kSetDialectNames]('db2', ['TIMESTAMP']);
+BaseTypes.STRING[kSetDialectNames]('db2', ['VARCHAR']);
+BaseTypes.CHAR[kSetDialectNames]('db2', ['CHAR']);
+BaseTypes.TEXT[kSetDialectNames]('db2', ['VARCHAR', 'CLOB']);
+BaseTypes.TINYINT[kSetDialectNames]('db2', ['SMALLINT']);
+BaseTypes.SMALLINT[kSetDialectNames]('db2', ['SMALLINT']);
+BaseTypes.MEDIUMINT[kSetDialectNames]('db2', ['INTEGER']);
+BaseTypes.INTEGER[kSetDialectNames]('db2', ['INTEGER']);
+BaseTypes.BIGINT[kSetDialectNames]('db2', ['BIGINT']);
+BaseTypes.FLOAT[kSetDialectNames]('db2', ['DOUBLE', 'REAL', 'FLOAT']);
+BaseTypes.TIME[kSetDialectNames]('db2', ['TIME']);
+BaseTypes.DATEONLY[kSetDialectNames]('db2', ['DATE']);
+BaseTypes.BOOLEAN[kSetDialectNames]('db2', [
+  'BOOLEAN',
+  'BOOL',
+  'SMALLINT',
+  'BIT',
+]);
+BaseTypes.BLOB[kSetDialectNames]('db2', ['BLOB']);
+BaseTypes.DECIMAL[kSetDialectNames]('db2', ['DECIMAL']);
+BaseTypes.UUID[kSetDialectNames]('db2', ['CHAR () FOR BIT DATA']);
+BaseTypes.ENUM[kSetDialectNames]('db2', ['VARCHAR']);
+BaseTypes.REAL[kSetDialectNames]('db2', ['REAL']);
+BaseTypes.DOUBLE[kSetDialectNames]('db2', ['DOUBLE']);
+BaseTypes.GEOMETRY[kSetDialectNames]('db2', false);
+
+export class BLOB extends BaseTypes.BLOB {
+  toSql() {
+    if (this._length) {
+      if (this._length.toLowerCase() === 'tiny') {
+        // tiny = 255 bytes
+        return 'BLOB(255)';
+      }
+
+      if (this._length.toLowerCase() === 'medium') {
+        // medium = 16M
+        return 'BLOB(16M)';
+      }
+
+      if (this._length.toLowerCase() === 'long') {
+        // long = 2GB
+        return 'BLOB(2G)';
+      }
+
+      return `BLOB(${this._length})`;
+    }
+
+    return 'BLOB'; // 1MB
+  }
+
+  static escape(blob: AcceptableTypeOf<BaseTypes.BLOB>, options: StringifyOptions) {
+    return `BLOB('${options.escape(blob.toString())}')`;
+  }
+
+  protected _stringify(value: AcceptableTypeOf<BaseTypes.BLOB>, options: StringifyOptions) {
+    if (Buffer.isBuffer(value)) {
+      return `BLOB('${options.escape(blob.toString())}')`;
+    }
+
+    if (Array.isArray(value)) {
+      value = Buffer.from(value);
+    } else {
+      value = Buffer.from(value.toString());
+    }
+
+    const hex = value.toString('hex');
+
+    return this._hexify(hex);
+  }
+
+  protected _hexify(hex: string) {
+    return `x'${hex}'`;
+  }
+}
+
+export class STRING extends BaseTypes.STRING {
+  static readonly escape: false;
```

Comment:
This is super spooky imo.

IIRC I determined this was safe, though I don't remember exactly why.

I'm not entirely certain that `_stringify` is always going to be called when formatting a user's input, but assuming it is this is safe?